zboxcore/sdk/writemarker_mutex.go

```diff
@@ -60,257 +52,215 @@ func CreateWriteMarkerMutex(client *client.Client, allocationObj *Allocation) (*
-	M := int(math.Ceil(float64(T) / float64(3) * float64(2))) //the minimum of M blobbers must accpet the marker
+	wg.Wait()
+}
 
-	//retry 3 times
-	for retry := 0; ; retry++ {
-		i := 1
-		n := 0
-		m.lockedBlobbers = nil
+// Change status code to 204
+func (wmMu *WriteMarkerMutex) UnlockBlobber(
+	ctx context.Context, b *blockchain.StorageNode,
+	connID string, timeOut time.Duration, wg *sync.WaitGroup,
+) {
 
-		body := url.Values{}
-		body.Set("connection_id", connectionID)
+	defer wg.Done()
 
-		now := time.Now()
-		body.Set("request_time", strconv.FormatInt(now.Unix(), 10))
+	var err error
 
-		form := body.Encode()
+	defer func() {
+		if err != nil {
+			logger.Logger.Error(err)
+		}
+	}()
 
-		for {
+	var req *http.Request
+	req, err = zboxutil.NewWriteMarkerUnLockRequest(
+		b.Baseurl, wmMu.allocationObj.Tx, connID, "")
+	if err != nil {
+		return
+	}
 
-			// M locks are acquired, it is safe to commit write
-			if n >= M {
-				return nil
-			}
+	var resp *http.Response
+	for retry := 0; retry < 3; retry++ {
+		if resp != nil {
+			resp.Body.Close()
+		}
 
-			// No more blobber, but n < M
-			if i > T && n < M {
-				//fails, release all locks
-				err := m.Unlock(ctx, connectionID)
-				if err != nil {
-					return err
-				}
-				break
-			}
+		reqCtx, ctxCncl := context.WithTimeout(ctx, timeOut)
+		resp, err = zboxutil.Client.Do(req.WithContext(reqCtx))
+		ctxCncl()
 
-			blobberIndex := i - 1
+		if err != nil {
+			return
+		}
+		if resp.StatusCode == http.StatusNoContent || resp.StatusCode == http.StatusOK {
+			logger.Logger.Info(b.Baseurl, connID, " unlocked")
+			return
+		}
 
-			result, err := m.lockOne(ctx, strings.NewReader(form), urls[blobberIndex])
+		if resp.StatusCode == http.StatusTooManyRequests {
+			logger.Logger.Info(
+				b.Baseurl, connID,
+				" got too many request error. Retrying")
 
-			// current blobber fails or is down, try next blobber
+			var r int
+			r, err = zboxutil.GetRateLimitValue(resp)
 			if err != nil {
-				// fails on current blobber
-				logger.Logger.Error("lock: ", err.Error())
-				i++
-				continue
+				return
 			}
 
-			// it is locked by other session, wait and retry
-			if result.Status == WMLockStatusPending {
-				logger.Logger.Info("WriteMarkerLock is pending, wait and retry")
-				sys.Sleep(1 * time.Second)
-				continue
-			} else if result.Status == WMLockStatusOK {
-				// locked on current blobber, count it and go to next blobber
-				if m.lockedBlobbers == nil {
-					m.lockedBlobbers = make(map[string]bool)
-				}
-
-				m.lockedBlobbers[m.allocationObj.Blobbers[blobberIndex].ID] = true
-				i++
-				n++
-			}
+			time.Sleep(time.Duration(r) * time.Second)
+			continue
+		}
+
+		var data []byte
+		data, err = io.ReadAll(resp.Body)
+		if err != nil {
+			return
 		}
 
-		if retry >= 2 {
-			return constants.ErrNotLockedWritMarker
+		m := string(data)
+		if m == "EOF" || strings.Contains(m, "server closed idle connection") {
```

Comment:
This 'server closed connection' error might be too specific, and the error could be changed as we change the ingress or upgrade the nginx. So perhaps we can just log the data directly and return instead of retry to avoid tight binding error messages.